salsa20/src/backends.rs

```diff
@@ -0,0 +1,10 @@
+use cfg_if::cfg_if;
+
+cfg_if! {
+    if #[cfg(any(target_arch = "x86", target_arch = "x86_64"))] {
+        pub(crate) mod sse2;
+        pub(crate) mod soft;
+    } else {
+        pub(crate) mod soft;
+    }
+}
```

Comment:
Seems like you don't need `cfg_if` anymore?

```suggestion
#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
pub(crate) mod sse2;
pub(crate) mod soft;
```